compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -93,6 +101,24 @@ pub(super) fn search_for_section<'a>(
         .map_err(|e| format!("failed to read {} section in '{}': {}", section, path.display(), e))
 }
 
+pub(super) fn get_metadata_xcoff<'a>(path: &Path, data: &'a [u8]) -> Result<&'a [u8], String> {
+    let Ok(file) = object::File::parse(data) else {
+        return Ok(bytes);
+    };
+    let buf = search_for_section(path, data, ".info")?;
+    let offset = file.symbol_address_by_name("rustc")?;
+    // The first 4 bytes from offset are length field for rustc metadata.
+    if offset + 4 >= buf.len() {
```

Comment:
This should be handled by `object` crate. For AIX toolchain, a `C_INFO` symbol's offset inside `.info` section doesn't count the first 4-bytes length field. When reading the object file, we can set `C_INFO` ObjectSymbol's address the offset into `.info` section and the size can be initialized from the first 4-bytes length field.